src/mds/Server.cc

```diff
@@ -6294,6 +6306,12 @@ void Server::handle_remove_vxattr(MDRequestRef& mdr, CInode *cur)
     // the rmxattr request to do this.
     handle_set_vxattr(mdr, cur);
     return;
+  } else if (name == "ceph.dir.pin"
+          || name == "ceph.dir.pin.distributed"
+          || name == "ceph.dir.pin.random"
+          || name == "ceph.dir.subvolume") {
+    handle_set_vxattr(mdr, cur);
```

Comment:
It seems the comment above this if block is no longer correct. How about we change the code to:

```
if (name.find("ceph.") == 0) {
  if (req->get_data().length()) {
     respond_to_request(mdr, -CEPHFS_EINVAL);
     return;
  }
  handle_set_vxattr(mdr, cur);
}
```

Reply:
Fix all vxattrs that should not allow removal to reject the RPC (EINVAL).

Reply:
That works for me. Thanks for the suggestion!

Reply:
@chrisphoffman don't forget:

> Fix all vxattrs that should not allow removal to reject the RPC (EINVAL).

Include in  your test that `setfattr -x ceph.X <path>`. fails for vxattrs where removal makes no sense. I think `ceph.quota` is another example of a vxattr which should support rmxattr.

Reply:
Took a while to unpack all of this. I did an audit of all xattrs handled. All ceph.* xattrs should either reset to def value, or reject for xattrs that shouldn't be removed (my judgement). If neither case, it is unknown xattr and will reject again with EINVAL.